scanner/src/main/kotlin/Scanner.kt

```diff
@@ -577,6 +587,41 @@ class Scanner(
         }
     }
 
+    private fun createMissingFileListings(controller: ScanController) {
+        val idsByProvenance = controller.getIdsByProvenance()
+        val provenancesMissingFileListings = idsByProvenance.keys.filterNot { fileListingResolver.has(it) }
+
+        logger.info { "Creating file listings for ${provenancesMissingFileListings.size} provenances." }
+
+        val duration = measureTime {
+            val previousIndex = AtomicInteger(0)
+
+            provenancesMissingFileListings.parallelStream().forEach { provenance ->
```

Comment:
As `parallelStream()` is limited to JVM targets, I'd prefer a more Kotlin-style approach with coroutines, similar to like recently done for the downloader CLI:

https://github.com/oss-review-toolkit/ort/blob/eb88bd776c0accb75f3741aaf398d858c4428739/cli/src/main/kotlin/commands/DownloaderCommand.kt#L312-L321

That'd also allow you to get rid of the manual counting with `AtomicInteger`.